Formula/qt5.rb

```diff
@@ -86,6 +87,7 @@ class Qt5 < Formula
   depends_on :mysql => :optional
   depends_on :postgresql => :optional
   depends_on :xcode => :build
+  depends_on "openssl" => :optional
```

Comment:
As far as I'm aware, yeah, there's no support for SecureTransport in either of those. Despite OpenSSL being deprecated on OS X/macOS for a fairly long time now a lot of upstreams don't seem tremendously interested in supporting yet _another_ crypto backend.

IMO, the `mysql` and `postgresql` options for `qt5` have always been more trouble than they are worth, but obviously people are using them and are determined to make them work come hell or high water. I know `postgresql` had a discussion upstream about supporting more crypto backends but I don't think SecureTransport support ever came out of that.

It's completely gross but personally I'd be tempted to make it:
```ruby
depends_on "openssl" if build.with?("postgresql") || build.with?("mysql")
```
That way you get the support without letting people deviate from the upstream SecureTransport default more than necessary.

The options here get kind of complicated by `mysql` and `postgresql` both being special dependencies and so not necessarily the Homebrew ones. The upstream pre-compiled `mysql` for example ships its own statically-linked `yassl` for the GPL version and `openssl` for the commercial version IIRC, so you could end up with Qt5 using an incompatible SSL compared to the dependency, but more or less that's a "user configuration" issue.